vck/src/commonTest/kotlin/at/asitplus/wallet/lib/data/rfc2045/MediaTypeTest.kt

```diff
@@ -0,0 +1,114 @@
+                        "APPLICATION;key=value",
+                    ),
+                    "application/jwt;key=value" to listOf(
+                        "application/jwt;key=value",
+                        "application/jWt;key=value",
+                        "application/JWT;key=value",
+                        "aPpLiCaTiOn/jwt;key=value",
+                        "aPpLiCaTiOn/jwt;key=value",
+                        "aPpLiCaTiOn/JWT;key=value",
+                        "APPLICATION/jwt;key=value",
+                        "APPLICATION/Jwt;key=value",
+                        "APPLICATION/JWT;key=value",
+                    ),
+                )
+            ) {
+                val original = testCase.name.testName
+                withData(it) {
+                    MediaType(original) shouldBe MediaType(it)
+                }
+            }
+            withData(
+                mapOf(
+                    "application;key=value" to listOf(
+                        "application;Key=value",
+                        "application;key=Value",
+                        "application;Key=Value",
+                        "aPpLiCaTiOn;Key=value",
+                        "aPpLiCaTiOn;key=Value",
+                        "aPpLiCaTiOn;Key=Value",
+                        "APPLICATION;Key=value",
+                        "APPLICATION;key=Value",
+                        "APPLICATION;Key=Value",
+                    ),
+                    "application/jwt;key=value" to listOf(
+                        "application/jwt;Key=value",
+                        "application/jwt;key=Value",
+                        "application/jwt;Key=Value",
+                        "application/jWt;Key=value",
+                        "application/jWt;key=Value",
+                        "application/jWt;Key=Value",
+                        "application/JWT;Key=value",
+                        "application/JWT;key=Value",
+                        "application/JWT;Key=Value",
+                        "aPpLiCaTiOn/jwt;Key=value",
+                        "aPpLiCaTiOn/jwt;key=Value",
+                        "aPpLiCaTiOn/jwt;Key=Value",
+                        "aPpLiCaTiOn/jwt;Key=value",
+                        "aPpLiCaTiOn/jwt;key=Value",
+                        "aPpLiCaTiOn/jwt;Key=Value",
+                        "aPpLiCaTiOn/JWT;Key=value",
+                        "aPpLiCaTiOn/JWT;key=Value",
+                        "aPpLiCaTiOn/JWT;Key=Value",
+                        "APPLICATION/jwt;Key=value",
+                        "APPLICATION/jwt;key=Value",
+                        "APPLICATION/jwt;Key=Value",
+                        "APPLICATION/Jwt;Key=value",
+                        "APPLICATION/Jwt;key=Value",
+                        "APPLICATION/Jwt;Key=Value",
+                        "APPLICATION/JWT;Key=value",
+                        "APPLICATION/JWT;key=Value",
+                        "APPLICATION/JWT;Key=Value",
```

Comment:
See comment below ... we'll need some method to generate mixedcased strings ...

```kotlin
.map { if (kotlin.random.Random.nextBoolean()) it.uppercaseChar() else it.lowercaseChar() }.joinToString("")
```

Reply:
Also, I think we're using ktor-http somewhere anyways ... maybe https://api.ktor.io/ktor-http/io.ktor.http/-content-type/index.html would be an option

Reply:
removed